From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TP2 BMC <tp2bmc@local>
Date: Mon, 02 Jun 2026 12:00:00 +0200
Subject: [PATCH 1/3] dt-bindings: i2c: add turing,i2c-smi-arbiter

Document the arbiter that serializes HW I2C (Profile A) and GPIO SMI
bitbang (Profile B) on shared TWI pins for Turing Pi 2.

Signed-off-by: TP2 BMC <tp2bmc@local>
---
 .../devicetree/bindings/i2c/turing,i2c-smi-arbiter.yaml | 28 ++++++++++++++++++++++
 1 file changed, 28 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/i2c/turing,i2c-smi-arbiter.yaml

diff --git a/Documentation/devicetree/bindings/i2c/turing,i2c-smi-arbiter.yaml b/Documentation/devicetree/bindings/i2c/turing,i2c-smi-arbiter.yaml
new file mode 100644
index 000000000000..a1b2c3d4e5f6
--- /dev/null
+++ b/Documentation/devicetree/bindings/i2c/turing,i2c-smi-arbiter.yaml
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/i2c/turing,i2c-smi-arbiter.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Turing Pi 2 I2C / SMI pin profile arbiter
+
+maintainers:
+  - Sven Rademakers <sven@turingpi.com>
+
+description: |
+  Serializes access to a shared two-wire bus where Profile A uses the SoC
+  hardware I2C controller (pinctrl "default") and Profile B uses GPIO SMI
+  bitbang (pinctrl "smi_mode") on the same pins.
+
+  The I2C controller node must define pinctrl-names = "default", "smi_mode".
+
+properties:
+  compatible:
+    const: turing,i2c-smi-arbiter
+
+  i2c-bus:
+    $ref: /schemas/i2c/i2c-controller.yaml#
+    description: Phandle to the shared I2C adapter (e.g. &i2c2).
+
+required:
+  - compatible
+  - i2c-bus
+
+additionalProperties: false
+
-- 
2.50.1
